Odata-docs/webapi-8/function-routing.md

```diff
@@ -0,0 +1,365 @@
+---
+title:  "Function Routing in ASP.NET Core OData 8"
+description: Function Routing in ASP.NET Core OData 8.
+date:   2022-12-07
+ms.date: 12/7/2022
+author: gathogojr
+ms.author: jogathog
+---
+
+# Function Routing in ASP.NET Core OData 8
+**Applies To**:[!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v8.md)]
+
+This tutorial shows how ASP.NET Core OData 8 supports function routing. A basic understanding of routing in ASP.NET Core OData 8 is assumed. If you're unfamiliar with routing in ASP.NET Core OData 8, you may want to go through the [routing overview](/odata/webapi-8/routing-overview) tutorial. You may also want to go through the [Actions & Functions](/odata/webapi-8/fundamentals/actions-functions) tutorial.
+
+## Introduction
+Functions are a way to add server-side logic that is not easily defined as CRUD (Create, Read, Update, and Delete) operations on entities. A function can target a single entity or a collection of entities. In OData terminology, this is _binding_. You can also have "unbound" functions, which are basically static operations on the OData service.
+
+OData function routing convention supports the following route templates:
+
+| Request Method | Route Template |
+|----------------|----------------|
+| `GET` | `~/{entityset}\|{singleton}/{function}` |
+| `GET` | `~/{entityset}\|{singleton}/{cast}/{function}` |
+| `GET` | `~/{entityset}/{key}/{function}` |
+| `GET` | `~/{entityset}/{key}/{cast}/{function}` |
+
+**Notes:**
+1. OData routing supports canonical parentheses-style key (e.g. `~/Employees(1)`) in addition to key-as-segment (e.g. `~/Employees/1`). Currently, ASP.NET Core OData 8 does not support key-as-segment convention in multi-part keys scenarios
+2. `{cast}` is a placeholder for the fully-qualified name for a derived type
+
+To illustrate function routing convention, let's build a sample OData service.
+
+## Prerequisites
+
+[!INCLUDE[](../includes/appliesto-webapi-v8-net-prereqs-vs.md)]
+
+## Packages
+
+[!INCLUDE[](../includes/appliesto-webapi-v8-pkg-install.md)]
+
+## Models
+The following are the models for the OData service. Define them under `FunctionRouting.Models` namespace:
```

Comment:
does this mean you set the `FunctionRouting.Models` namespace? I think it is confusing.. The statement could be left out.

Reply:
This is the namespace for the C# classes. It's mentioned under the **Models** section so it's relevant to the data models. The section for building Edm model is different and we don't change the Edm model namespace so it'll assume the value of `Default`.

Reply:
@ElizabethOkerio Would it help if I removed that statement and then replaced this:
```csharp
public class Employee
{
    public int Id { get; set; }
    public string Name { get; set; }
    public int PerfRating { get; set; }
}
```
with this:
```
namespace FunctionRouting.Models
{
    public class Employee
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int PerfRating { get; set; }
    }
}
```
?
I had added that statement because there's a review comment where you had said I should mention about the namespace. This second way may be less confusing but someone will still have to understand that this namespace finds it's way to the Edm model.
I think for now I'll implement my suggestion here since it needs less explanation